large_language_model/megatron-lm/README.md

```diff
@@ -202,3 +160,85 @@ Evaluate after every 24576 samples (=50.33B tokens)
 ### Evaluation thoroughness
 Evaluation on the validation subset that consists of 24567 examples.
 
+
+# 6. Other
+
+### S3 artifacts download
+The dataset and the checkpoints are available to download from an S3 bucket.
+To achieve the best download bandwidth (currently no more than 25MB/s is expected) it's necessary to set up a third-party client capable of downloading the artifacts.
+[Here are the instructions](https://help.lyvecloud.seagate.com/en/connecting-s3-clients-to-lyve-cloud.html).
+The read-only access credentials are provided below.
+
+#### Access details
+- Bucket name: `mlcommons-training-wg-s3`
+- Endpoint URL: https://s3.us-east-1.lyvecloud.seagate.com
+- Access Key: `3ZC41B4Z2WHM5DT2`
+- Secret Key: `AK4NQQZV0NKFEJWJUZVPX5XQ0QNTXCGW`
+
+
+### Model conversion from Paxml checkpoints
+Alternatively to downloading the checkpoint in Megatron ready format, it can be obtained by converting a Paxml checkpoint.
+
+Paxml Checkpoint is available at: `gs://mlperf-llm-public2/gpt3_spmd1x64x24_tpuv4-3072_v84_20221101/checkpoints/checkpoint_00004000`
+To resume training from the above checkpoint on Megatron, it should be converted into a format suitable for Megatron (this step only needs to be done once).
+
+To convert Paxml checkpoint to the Megatron's format, a [script](scripts/convert_paxml_to_megatron_distributed.py) has been provided:
+```bash
+# Convert model and optimizer parameters to Megatron format (runs in ~40 minutes on DGXA100, requires 1TB of CPU memory):
+python -u convert_paxml_to_megatron_distributed.py -gckpt $PAXML_CKPT_PATH -o $EXTERNAL_MODEL_CHECKPOINT_DIR --dtype fp32  # or `--dtype bf16` for BF16 checkpoint
+# Add framework-specific common.pt file to the checkpoint (instantaneous):
+python json_to_torch.py -i common_fp32.json -o $EXTERNAL_MODEL_CHECKPOINT_DIR/common.pt  # or `-i common_bf16.json` for BF16 checkpoint
+```
+This should result in the same checkpoint as described in the "Checkpoint download" section above.
+
+### Dataset preprocessing
+Here are the instructions to prepare the preprocessed dataset from scratch.
+
+#### Data Download
+Training dataset -
+```bash
+GIT_LFS_SKIP_SMUDGE=1 git clone https://huggingface.co/datasets/allenai/c4
+mkdir <${C4_PATH}>
+cd <${C4_PATH}>
+git lfs pull --include "en/c4-train.007*.json.gz"
+git lfs pull --include "en/c4-train.008*.json.gz"
+git lfs pull --include "en/c4-train.009*.json.gz"
+git lfs pull --include "en/c4-train.01*.json.gz"
+```
+
+Validation dataset needs to be downloaded from `gs://mlperf-llm-public2/c4/en_val_subset_json/c4-validation_24567exp.json` to ${C4_PATH}.
+
+#### Data Preprocessing for Megatron-LM
```

Comment:
Maybe rename this section to `Data Preprocessing for Megatron-LM from scratch` so it is clearer?